src/abi/json-abi-format.md

```diff
@@ -35,52 +36,36 @@ The ABI of a contract is represented as a JSON object containing the following p
-  - `"output"`: an object representing the output of the function (i.e. its return value). The output is a _type application_, which is a JSON object that contains the following properties:
-    - `"type"`: the _type declaration_ hash based ID of the type of the output.
-    - `"typeArguments"`: an array of the _type arguments_ used when applying the type of the output, if the type is generic, and `null` otherwise. Each _type argument_ is a _type application_ represented as a JSON object that contains the following properties:
-      - `"type"`: the _type declaration_ hash based ID of the type of the _type argument_.
-      - `"typeArguments"`: an array of the _type arguments_ used when applying the type of the _type argument_, if the type is generic, and `null` otherwise. The format of the elements of this array recursively follows the rules described in this section.
-  - `"attributes"`: an optional array of _attributes_. Each _attribute_ is explained in the [dedicated section](#attributes-semantics) and is represented as a JSON object that contains the following properties:
-    - `"name"`: the name of the attribute.
-    - `"arguments"`: an array of attribute arguments.
+    - `"concreteTypeId"`: the _type concrete declaration_ hash based ID of the type of the input.
+  - `"output"`: the _type concrete declaration_ hash based ID of the type being returned by the function.
 - `"loggedTypes"`: an array describing all instances of [`log`](../fuel-vm/instruction-set.md#log-log-event) or [`logd`](../fuel-vm/instruction-set.md#logd-log-data-event) in the contract's bytecode. Each instance is a JSON object that contains the following properties:
   - `"logId"`: a string containing the 64bit hash based decimal ID calculated from the first 8 bytes of the `sha256` of a string that represents the type logged as defined in [Hash Based Ids](./hash-based-ids.md). The [`log`](../fuel-vm/instruction-set.md#log-log-event) and [`logd`](../fuel-vm/instruction-set.md#logd-log-data-event) instructions must set their `$rB` register to that ID.
-  - `"loggedType"`: a _type application_ represented as a JSON object that contains the following properties:
-    - `"type"`: the _type declaration_ hash based ID of the type of the value being logged.
-    - `"typeArguments"`: an array of the _type arguments_ used when applying the type of the value being logged, if the type is generic, and `null` otherwise. Each _type argument_ is a _type application_ represented as a JSON object that contains the following properties:
-      - `"type"`: the _type declaration_ hash based ID of the type of the _type argument_.
-      - `"typeArguments"`: an array of the _type arguments_ used when applying the type of the _type argument_, if the type is generic, and `null` otherwise. The format of the elements of this array recursively follows the rules described in this section.
+  - `"loggedType"`: the _type concrete declaration_ hash based ID of the value being logged.
```

Comment:
I think renaming this to `concreteTypeId` to follow the same pattern as function inputs makes sense.